### PR TITLE
Add modified flag in Metadatas and remove canUseStatistics()

### PR DESCRIPTION
**Pull request by wshao08**
This PR adds a boolean flag in TimeseriesMetadata, ChunkMetadata and PageHeader to identify if their data is integral or modified.
Previously we use canUseStatistics() in Statistics for this reason, therefore remove it in this PR.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 1]
```diff
@@ -402,6 +410,13 @@ Statistics currentPageStatistics() {
     return firstPageReader.getStatistics();
   }
 
+  boolean currentPageModified() {
+    if (firstPageReader == null) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:414]: if firstPageReader is null, throw exception

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 2]
```diff
@@ -266,6 +270,10 @@ Statistics currentChunkStatistics() {
     return firstChunkMetadata.getStatistics();
   }
 
+  boolean currentChunkModified() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:273]: if firstChunkMetadata is null, throw exception

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 3]
```diff
@@ -184,6 +184,10 @@ Statistics currentFileStatistics() {
     return firstTimeSeriesMetadata.getStatistics();
   }
 
+  boolean currentFileModified() {
+    return firstTimeSeriesMetadata.hasModification();
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:188]: if firstTimeSeriesMetadata is null, throw exception

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 4]
```diff
@@ -693,5 +708,9 @@ BatchData getAllSatisfiedPageData() throws IOException {
     void setFilter(Filter filter) {
       data.setFilter(filter);
     }
+
+    boolean hasModification() {
+      return data.hasModification();
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:713]:
```suggestion
      return pageReader.hasModification();
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/header/PageHeader.java [hunk 5]
```diff
@@ -33,6 +33,7 @@
   private int uncompressedSize;
   private int compressedSize;
   private Statistics statistics;
+  private boolean pageModified;
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/header/PageHeader.java:36]:
```suggestion
  private boolean modified;
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetadata.java [hunk 6]
```diff
@@ -53,6 +53,8 @@
    */
   private long deletedAt = Long.MIN_VALUE;
 
+  private boolean chunkModified;
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetadata.java:56]:
```suggestion
  private boolean modified;
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java [hunk 7]
```diff
@@ -39,6 +39,7 @@
   
   private Statistics<?> statistics;
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java:41]:
```suggestion
// modified is true when there are modifications of the series, or from unseq file
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java [hunk 8]
```diff
@@ -116,4 +117,12 @@ public void setChunkMetadataLoader(IChunkMetadataLoader chunkMetadataLoader) {
   public List<ChunkMetadata> loadChunkMetadataList() throws IOException {
     return chunkMetadataLoader.loadChunkMetadataList();
   }
+
+  public boolean hasModification() {
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java:121]:
```suggestion
  public boolean isModified() {
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java [hunk 9]
```diff
@@ -416,14 +416,6 @@ public void setCount(long count) {
     this.count = count;
   }
 
-  public boolean canUseStatistics() {
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java:419]: remove the  canUseStatistics field in this class

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/IPageReader.java [hunk 10]
```diff
@@ -31,4 +31,6 @@
   Statistics getStatistics();
 
   void setFilter(Filter filter);
+
+  boolean hasModification();
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/IPageReader.java:35]:
```suggestion
  boolean isModified();
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java (hunk 1 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:420]: fixed

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java (hunk 2 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java (hunk 3 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java]: fixed

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java (hunk 9 above)
Comment (wshao08) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java:419]: fixed

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java (hunk 8 above)
Comment (wshao08) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TimeseriesMetadata.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/IPageReader.java (hunk 10 above)
Comment (wshao08) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/IPageReader.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/header/PageHeader.java (hunk 5 above)
Comment (wshao08) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/header/PageHeader.java]: done

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetadata.java (hunk 6 above)
Comment (wshao08) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetadata.java]: fixed

**Pull request merged**